internal/pkg/service/appsproxy/syncmap/syncmap_test.go

```diff
@@ -0,0 +1,40 @@
+package syncmap_test
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/atomic"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/appsproxy/syncmap"
+)
+
+type testStruct struct{}
+
+func TestSyncMap(t *testing.T) {
+	t.Parallel()
+
+	m := syncmap.NewSyncMap[string, testStruct](func() *testStruct {
+		return &testStruct{}
+	})
+
+	wg := sync.WaitGroup{}
+	counter := atomic.NewInt64(0)
+	for i := 0; i < 10; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			m.GetOrInit("test")
```

Comment:
I'm missing an assert here, e.g. that you got the same reference 10 times.

Reply:
Ok, so it tests the race condition, and it doesn't test the internal function of that map.

The easiest is to do 2 separate tests (so you can add another one).
 - `TestSyncMap_GetOrInit_Race`
 - `TestSyncMap_GetOrInit`